network_manager/src/network_manager.cpp

```diff
@@ -0,0 +1,155 @@
+#include "network_manager.hpp"
+
+#include <string.h>
+#include <regex>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+#include "lwip/inet.h"
+#include "lwip/sockets.h"
+#include "esp_sntp.h"
+
+namespace sdk
+{
+    res network_manager::get_status()
+    {
+        return Ok(RUNNING);
+    }
+
+    res network_manager::run()
+    {
+        return Ok(RUNNING);
+    }
+
+    res network_manager::stop()
+    {
+        auto sta_ret = sta.stop();
+        auto ap_ret = ap.stop();
+        if (sta_ret.isErr() && ap_ret.isErr())
+        {
+            auto string = etl::make_string_with_capacity<128>("Error stopping Wifi. STA: ");
+            string += sta_ret.unwrapErr();
+            string += etl::make_string(" AP: ");
```

Comment:
Pedantic, but having the "AP:" like `"\n\tAP:\n"` might be easier to read if we do need these logs at some point.

Reply:
Also updating the component status and returning it when `run()` is called might be helpful for a recovery flow. For example when STA encounters an `asoc` error, that would allow the manager to help you retry the connection.